tests/application_development/libcxx/testcase.yaml

```diff
@@ -2,11 +2,13 @@ tests:
   application_development.cpp.libcxx:
     platform_exclude: qemu_x86_coverage
     toolchain_exclude: xcc
+    filter: TOOLCHAIN_HAS_NEWLIB == 1
```

Comment:
Just as a note: technically this isn't right, since what we really require is a libcxx, whether that comes from newlib or not.  Ultimately this should be `TOOLCHAIN_HAS_LIBCXX == ` but for now this is OK.